app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/model/CafeteriaMenu.kt

```diff
@@ -25,4 +27,63 @@ data class CafeteriaMenu(@PrimaryKey(autoGenerate = true)
                          var typeShort: String = "",
                          var typeLong: String = "",
                          var typeNr: Int = -1,
-                         var name: String = "")
\ No newline at end of file
+                         var name: String = "") {
+
+    private val formattedName: String
+        get() = COMPILE.matcher(name).replaceAll("").trim()
+
+    val menuType: MenuType
+        get() {
+            return when (typeShort) {
+                "tg" -> MenuType.DAILY_SPECIAL
+                "ae" -> MenuType.DISCOUNTED_COURSE
+                "akt" -> MenuType.SPECIALS
+                "bio" -> MenuType.BIO
+                else -> MenuType.SIDE_DISH
+            }
+        }
+
+    val notificationTitle: String
+        get() {
+            return PATTERN
+                    .matcher(typeLong)
+                    .replaceAll("")
+                    .trim()
+        }
+
+    fun getNotificationText(context: Context): String {
+        val lines = getNotificationLines(context)
+        return if (menuType == MenuType.SPECIALS) {
+            lines.joinToString(", ")
```

Comment:
this should be `getNotificationLines(context).apply(...)`

Reply:
Why?

Reply:
I think it's more readable, but I might also be overusing it ... might boil down to preference
```kotlin
return getNotificationLines(context).apply {
    if(menuType == MenuType.SPECIALS) {
        joinToString(", ")
    } else {
        first()
    }
}
```

Reply:
This would return the result of `getNotificationLines()`, not the joined string or the first list element. `run { }` might be an alternative, but at this point, I think we’re over-engineering this small method.